transports/p2p/connmgr/connmanager.go

```diff
@@ -209,17 +253,25 @@ func (cm *ConnManager) handleFailedConn(c *ConnReq) {
 			cm.Connect(c)
 		})
 	} else if cm.cfg.GetNewAddress != nil {
-		cm.failedAttempts++
-		if cm.failedAttempts >= maxFailedAttempts {
-			cm.log.Debug().Msgf("Max failed connection attempts reached: [%d] "+
-				"-- retrying connection in: %v", maxFailedAttempts,
-				cm.cfg.RetryDuration)
-			time.AfterFunc(cm.cfg.RetryDuration, func() {
-				cm.NewConnReq()
-			})
+		if c.Addr != nil && c.Addr.String() != "" && cm.cfg.BanAddress != nil {
+			cm.incAddrConnAttempt(c.Addr.String())
+			if cm.getAddrConnAttempt(c.Addr.String()) >= maxFailedAttempts {
+				cm.cfg.BanAddress(c.Addr.String())
```

Comment:
I'm worrying that this function `BanAddress` could be used by multiple goroutines and it is not thread safe because it utilizes a standard `map`.

Maybe my concerns are pointless, if so - the function will work completely fine.

Reply:
Okay, I used the AddrManager mutex, so it should be fine now